app/javascript/react/screens/App/Overview/screens/MappingWizard/components/MappingWizardDatastoresStep/components/DatastoresStepForm.js

```diff
@@ -49,28 +52,93 @@ class DatastoresStepForm extends React.Component {
+          targetDatastoreWithSourceDatastores =>
+            targetDatastoreWithSourceDatastores.id ===
+            selectedTargetDatastore.id
+        )
       );
-      if (isAlreadySelected) {
+
+    const isNotAlreadySelected = !selectedSourceDatastores.some(datastore => {
+      return datastore.id === sourceDatastore.id;
+    });
+
+    if (
+      selectedTargetDatastore &&
+      mappingExistsForTargetDatastore &&
+      isNotAlreadySelected &&
+      targetDatastoreAvailableSpace(
+        selectedTargetDatastore,
+        datastoresStepMappings
+      ) < totalUsedSpace([...selectedSourceDatastores, sourceDatastore])
+    ) {
+      showAlertAction(errorMessage);
+    } else if (
+      selectedTargetDatastore &&
+      isNotAlreadySelected &&
+      selectedTargetDatastore.free_space <
+        totalUsedSpace([...selectedSourceDatastores, sourceDatastore])
+    ) {
+      showAlertAction(errorMessage);
+    } else {
+      this.setState(prevState => {
+        const isAlreadySelected = prevState.selectedSourceDatastores.some(
+          datastore => datastore.id === sourceDatastore.id
+        );
+        if (isAlreadySelected) {
+          return {
+            selectedSourceDatastores: prevState.selectedSourceDatastores.filter(
+              datastore => datastore.id !== sourceDatastore.id
+            )
+          };
+        }
+        hideAlertAction();
         return {
-          selectedSourceDatastores: prevState.selectedSourceDatastores.filter(
-            datastore => datastore.id !== sourceDatastore.id
-          )
+          selectedSourceDatastores: [
+            ...prevState.selectedSourceDatastores,
+            sourceDatastore
+          ]
         };
-      }
-      return {
-        selectedSourceDatastores: [
-          ...prevState.selectedSourceDatastores,
-          sourceDatastore
-        ]
-      };
-    });
+      });
+    }
   }
 
   selectTargetDatastore(targetDatastore) {
-    this.setState(() => ({ selectedTargetDatastore: targetDatastore }));
+    const { selectedSourceDatastores } = this.state;
+    const { value: datastoresStepMappings } = this.props.input;
+    const { showAlertAction, hideAlertAction } = this.props;
+
+    const mappingExistsForTargetDatastore = datastoresStepMappings.some(
+      targetClusterWithDatastoreMappings =>
+        targetClusterWithDatastoreMappings.nodes.some(
+          targetDatastoreWithSourceDatastores =>
+            targetDatastoreWithSourceDatastores.id === targetDatastore.id
+        )
+    );
+
+    if (
+      selectedSourceDatastores.length > 0 &&
+      mappingExistsForTargetDatastore &&
+      targetDatastoreAvailableSpace(targetDatastore, datastoresStepMappings) <
+        totalUsedSpace(selectedSourceDatastores)
+    ) {
+      showAlertAction(errorMessage);
+    } else if (
+      selectedSourceDatastores.length > 0 &&
+      targetDatastore.free_space < totalUsedSpace(selectedSourceDatastores)
+    ) {
```

Comment:
It's fine for now, but you basically have here: `if (a) { doThing(); } else if (b) { doThing(); } else { ... }`, where it could be `if(a || b) { doThing(); } else { ... }`. We could assign these conditions to variables for readability.

Not a blocker, looks fine for now.